units/stats_task_end.go

```diff
@@ -104,6 +105,15 @@ func (j *collectTaskEndDataJob) Run(ctx context.Context) {
 		j.env = evergreen.GetEnvironment()
 	}
 
+	pRef, err := model.FindOneProjectRef(j.task.Project)
+	var projectIdentifier string
+	if pRef == nil {
+		projectIdentifier = "not_found"
```

Comment:
An alternative would be to set it after creating the `msg`.

```
if pRef != nil {
   msg["project_identifier"] = pRef
}
```

I don't think it's more correct to do it that way, but it is a more common approach in our codebase and takes less code. On the other hand, it puts the nil check later, which is maybe more confusing. I'm open to either, I think, since I can't think of a strong argument either way.